daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/PrimitivesTextNumber.scala

```diff
@@ -114,21 +191,66 @@ object TextNumberPatternUtils {
    * Checks a pattern for suitability with V (virtual decimal point) in the pattern
    * in the context of zoned textNumberRep.
    *
-   * This is broken out separately for unit testing purposes.
+   * This method is here in this object for unit testing purposes.
    *
    * @param patternStripped the dfdl:textNumberPattern pattern - with all quoting removed.
    * @return None if the pattern is illegal syntax for use with V (virtual decimal point)
    *         otherwise Some(N) where N is the number of digits to the right of the V character.
    */
-  private[primitives] def textDecimalVirtualPointForZoned(patternStripped: String): Option[Int] = {
-    val r = TextNumberPatternUtils.vregexZoned
+  def textNumber_V_DecimalVirtualPointForZoned(patternStripped: String): Option[Int] = {
```

Comment:
Since this is no longer `private[primitives]`, does it make sense to move the regex match logic to the `textDecimalVirtualPointFromPattern` val in `PrimivitesZoned.scala`? That would make the different `textDecimalVirtualPointFromPattern` implementations look more similar instead of one calling this function and the other having the logic inline.